### PR TITLE
test(step-sequence): playwright refactor

### PR DESCRIPTION
**Pull request by divyajindel**
### Proposed behaviour

- Copy `step-sequence` folder from `/cypress/locators/` to `/playwright/components/`
- Rename `index.js` to `index.ts` and `locators.js` to `locators.ts` in `/playwright/components/step-sequence/` folder
- Refactor `index.ts` to Playwright
- Copy `step-sequence.cy.tsx from `/cypress/components/step-sequence/` to `/src/components/step-sequence/`
- Rename `step-sequence.cy.tsx` to `step-sequence.pw.tsx` in `/src/components/step-sequence/` folder
- Refactor tests to Playwright

### Current behaviour

`Step-sequence` tests are in Cypress not Playwright.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions
- [x] Run step-sequence tests in Playwright Test Runner to check if the `step-sequence.pw.tsx` file passed
- [x] Run all other tests in Playwright Test Runner to check none of the other `*.pw.tsx` files have regressed

### CODE REVIEW

File: src/components/step-sequence/step-sequence.pw.tsx [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+          color
+        );
+      });
+    }
+  );
+});
+
+test.describe("Accessibility tests for StepSequence component", () => {
+  ["horizontal", "vertical"].forEach((orientation) => {
+    test(`should check orientation set to ${orientation}`, async ({
```
Comment (DipperTheDan) [inline src/components/step-sequence/step-sequence.pw.tsx:180]:
```suggestion
    test(`should check orientation is set to ${orientation}`, async ({
```

File: src/components/step-sequence/step-sequence.pw.tsx [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+      mount,
+      page,
+    }) => {
+      await mount(<StepSequenceComponent orientation={orientation} />);
+
+      await checkAccessibility(page);
+    });
+  });
+
+  test("should check children for accessibility tests", async ({
```
Comment (DipperTheDan) [inline src/components/step-sequence/step-sequence.pw.tsx:190]: Or something like that. 

```suggestion
  test("should check children for StepSequenceItemCustom example", async ({
```

File: src/components/step-sequence/step-sequence.pw.tsx [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+    mount,
+    page,
+  }) => {
+    await mount(<StepSequenceItemCustom />);
+
+    await checkAccessibility(page);
+  });
+
+  ["-100", "0", "999", testData[0], testData[1]].forEach((indicator) => {
+    test(`should check indicator set to ${indicator}`, async ({
```
Comment (DipperTheDan) [inline src/components/step-sequence/step-sequence.pw.tsx:200]:
```suggestion
    test(`should check indicator is set to ${indicator}`, async ({
```

File: src/components/step-sequence/step-sequence.pw.tsx [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+    }) => {
+      await mount(
+        <StepSequenceItemCustom status="incomplete" indicator={indicator} />
+      );
+      await checkAccessibility(page);
+    });
+  });
+
+  testData.forEach((ariaLabel) => {
+    test(`should check ariaLabel set to ${ariaLabel}`, async ({
```
Comment (DipperTheDan) [inline src/components/step-sequence/step-sequence.pw.tsx:212]:
```suggestion
    test(`should check ariaLabel is set to ${ariaLabel}`, async ({
```

File: src/components/step-sequence/step-sequence.pw.tsx [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+      await checkAccessibility(page);
+    });
+  });
+
+  ([
+    "complete",
+    "current",
+    "incomplete",
+  ] as StepSequenceItemProps["status"][]).forEach((status) => {
+    test(`should check status set to ${status}`, async ({ mount, page }) => {
```
Comment (DipperTheDan) [inline src/components/step-sequence/step-sequence.pw.tsx:226]:
```suggestion
    test(`should check status is set to ${status}`, async ({ mount, page }) => {
```

File: src/components/step-sequence/step-sequence.pw.tsx [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+      await mount(<StepSequenceItemCustom status={status} />);
+      await checkAccessibility(page);
+    });
+  });
+
+  testData.forEach((hiddenCompleteLabel) => {
+    test(`should check hiddenCompleteLabel set to ${hiddenCompleteLabel}`, async ({
```
Comment (DipperTheDan) [inline src/components/step-sequence/step-sequence.pw.tsx:233]:
```suggestion
    test(`should check hiddenCompleteLabel is set to ${hiddenCompleteLabel}`, async ({
```

File: src/components/step-sequence/step-sequence.pw.tsx [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+          status="complete"
+          hiddenCompleteLabel={hiddenCompleteLabel}
+        />
+      );
+      await checkAccessibility(page);
+    });
+  });
+
+  testData.forEach((hiddenCurrentLabel) => {
+    test(`should check hiddenCurrentLabel set to ${hiddenCurrentLabel}`, async ({
```
Comment (DipperTheDan) [inline src/components/step-sequence/step-sequence.pw.tsx:248]:
```suggestion
    test(`should check hiddenCurrentLabel is set to ${hiddenCurrentLabel}`, async ({
```

File: src/components/step-sequence/step-sequence.pw.tsx [hunk 8]
```diff
@@ -0,0 +1,275 @@
...
+      await checkAccessibility(page);
+    });
+  });
+
+  ([
+    "complete",
+    "current",
+    "incomplete",
+  ] as StepSequenceItemProps["status"][]).forEach((status) => {
+    test(`should check hideIndicator prop when status set to ${status}`, async ({
```
Comment (DipperTheDan) [inline src/components/step-sequence/step-sequence.pw.tsx:267]:
```suggestion
    test(`should check hideIndicator prop when status is set to ${status}`, async ({
```

**Review by DipperTheDan — DISMISSED** [review]

File: src/components/step-sequence/step-sequence.pw.tsx [hunk 9]
```diff
@@ -0,0 +1,278 @@
...
+        />
+      );
+      const expectedLabelChild = stepSequenceDataComponentItem(page)
+        .locator("span")
+        .nth(0);
+      await expect(expectedLabelChild).toHaveText(hiddenCurrentLabel);
+    });
+  });
+
+  test("should check hideIndicator prop when status is is set to complete", async ({
```
Comment (sianford) [inline src/components/step-sequence/step-sequence.pw.tsx:143]: Extra word "is" to remove.

**Review by DipperTheDan — APPROVED** [review]

**Pull request merged**